api/slack.go

```diff
@@ -6,24 +6,31 @@ import (
 	"github.com/slack-go/slack"
 )
 
-func SendSlackMessages(slackToken string, messages map[string]string) {
-	log := logger.Get()
+type SlackClientInterface interface {
+	PostMessage(channelID string, options ...slack.MsgOption) (string, string, error)
+}
 
-	if len(slackToken) > 0 {
-		slackClient := slack.New(slackToken, slack.OptionDebug(true))
-		for channel, message := range messages {
-			_, timestamp, err := slackClient.PostMessage(
-				channel,
-				slack.MsgOptionText(message, false),
-				slack.MsgOptionAsUser(true),
-			)
+type SlackClient struct {
+	client SlackClientInterface
+}
+
+func (s *SlackClient) PostMessage(channelID string, options ...slack.MsgOption) (string, string, error) {
+	return s.client.PostMessage(channelID, options...)
+}
```

Comment:
Since the struct returned by `slack.New` has a `PostMessage` receiver, that should satisfy the interface of `SlackClientInterface`, right? So why add this extra wrapper?

Reply:
This actually is necessary. If the PostMessage method is removed from the SlackClient struct, the SlackClient will no longer satisfy the SlackClientInterface interface, as it will not implement the PostMessage method required by the interface. As a result, when calling the NewSlackClient function, which returns an instance of SlackClient, it would not satisfy the SlackClientInterface interface, and any code that relies on the SlackClientInterface interface would not work correctly. Let me know If that makes sense!

Reply:
Right, I'm wondering though if the `SlackClient` struct itself is necessary. The definition of the `SlackClientInterface` requires only that the implementation has a receiver for `PostMessage`. Since the implementation returned by `slack.New` has a receiver for `PostMessage`, that means it directly satisfies the interface.

Reply:
Yeah, I went with this approach because it was simple to mock and write unit tests for using the PostMessage implementation. But I'll see if I can find another way that doesn't rely on PostMessage.